vllm/worker/vineyard_llm_cache.py

```diff
@@ -165,8 +163,7 @@ def prefetch_seq_kv_caches(
 
         # synchronized across tensor parallel ranks
         matched_tensor = torch.tensor([matched], dtype=torch.long, device='cuda')
-        # torch.distributed.all_reduce(matched_tensor, op=torch.distributed.ReduceOp.MIN,
-        #                              group=get_tensor_model_parallel_group())
+        get_tp_group().all_reduce(input_=matched_tensor, op=torch.distributed.ReduceOp.MIN)
```

Comment:
So actually every worker has to perform cache query and then negotiates the least `matched`